### PR TITLE
User profile: basic info, course list, projects

### PR DESCRIPTION
**Pull request by katiearriagam**
# Changes 
This PR includes the changes necessary for the student profile view. Includes:
- User's basic info 
- User's projects
- User's courses (and functionality to include a new course)
- Tested responsiveness

## Changes per file
In `UserController.php`, I added:
- Missing const 
- Server side HTTP status error code (client is not at fault here)
- Error handling to prevent registration of duplicate student-course relationships.
- `associate()` method now has teachers as part of the course.

In `app.scss`: 
- Fixed responsiveness for project views 

In `projectCard.blade.php`:
- Check for set `skillsets`. We are not including these in the user's profile since they clutter the view and do not seem to add value here because the view is a portfolio, and not a project browsing view. 

In `projects.blade.php`:
- Fixed responsiveness

In `details.blade.php`:
- Renamed file from `\resources\views\user\studentProfile.blade.php` to `\resources\views\user\details.blade.php` (did this with mv command on [this commit](https://github.com/ProyectoIntegrador2018/BlueMarket/pull/153/commits/ea236a117cdccc850f60facf3cecbb8d0ebedd6c)).
- Defined user profile responsive layout
- Added basic info to user profile 
- `Contact` button in user profile
- `Edit` button in user profile (only shows when user is looking at their own profile)
- Added course list in user profile, as well as possibility to add a new course if user is looking at their own profile. 

# Missing (expect these in a future PR)
- Placeholder image for users that do not have an image set.
- List of user's teams w/link to the team profile. 
 Note: @hecerinc @anakarenbm: creo que ahorita no tengo forma de acceder projects que se relacionen con un usuario. ¿Hay algún método que pueda usar que exponga los projects del user (o al menos de los que sus teams son owners)? 

# Related
- Partially fixes: #121 
- Fixes: #152 

# Current views

## Course list
image

## Course registration workflow
Looks mostly the same... casi todos los cambios son limpieza de lo que teníamos antes.

image

image

image

## Projects list
image

## Responsiveness
image

image

image

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -101,14 +103,19 @@ public function getCourseDetails(Request $request) {
 	public function associate(Request $request) {
 		$user = Auth::user();
 		$courseKey = $request->courseKey;
-		$course = Course::where('course_key', $courseKey)->first();
+		$course = Course::with('teachers')->where('course_key', $courseKey)->first();
+
+		abort_if($course == null || $user == null, 509);
+		abort_if($user->role != config(self::ROLES)['student'], 509);
```
Comment (hecerinc) [inline app/Http/Controllers/UserController.php]: `enum.user_roles` will do here. No hay necesidad de declarar la constante.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -101,14 +103,19 @@ public function getCourseDetails(Request $request) {
...
+
+		// prevent insertion of duplicates
+		$associatedCourse = $user->enrolledIn()->where('course_id', $course->id)->first();
```
Comment (hecerinc) [inline app/Http/Controllers/UserController.php]: `exists` es el m&eacute;todo si s&oacute;lo quieres checar que exista.

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -101,14 +103,19 @@ public function getCourseDetails(Request $request) {
 	public function associate(Request $request) {
 		$user = Auth::user();
 		$courseKey = $request->courseKey;
-		$course = Course::where('course_key', $courseKey)->first();
+		$course = Course::with('teachers')->where('course_key', $courseKey)->first();
+
+		abort_if($course == null || $user == null, 509);
```
Comment (hecerinc) [inline app/Http/Controllers/UserController.php]: 509 Bandwidth Limit Exceeded ?

File: resources/views/user/details.blade.php [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+		<div class="ui stackable grid">
+			<div class="four wide column user-basic-info">
+				<!-- User avatar -->
+				<img class="ui small circular image user-avatar" src="<?php echo asset($user->picture_url) ?>"/>
+				<!-- User name -->
+				<h2 class="center" style="text-align: center;">{{ $user->name }}</h2>
+				<!-- Contact button -->
+				<a class="fluid ui primary button buttonSpace" href="mailto:{{ $user->email }}" style="margin-bottom: 10px;"><i class="icon envelope"></i>Contact</a>
+				<!-- Edit button -->
+				@if ($user->id == Auth::user()->id)
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]: `Auth::id()`

File: resources/views/user/details.blade.php [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+						<div id="no-projects-msg" class="ui message">
+							<div class="header">
+								No projects found!
+							</div>
+							<p>Looks like this user is not participating in any projects... yet!</p>
+						</div>
+					@endif
+				</div>
+				<div class="ui bottom attached tab segment" data-tab="second">
+					Second
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]: Is this supposed to be like this?

File: app/Http/Controllers/UserController.php (hunk 3 above)
Comment (hecerinc) [reply app/Http/Controllers/UserController.php]: Probably should send back an actual message to eh user that the course key is incorrect.

File: resources/views/user/details.blade.php [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+				</div>
+				<div class="ui bottom attached tab segment" data-tab="courses">
+					<div class="courses">
+						@if($user->id == Auth::user()->id)
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]: `Auth::id()`

File: resources/views/user/details.blade.php [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+@section("scripts")
+<script>
+	/* Semantic UI setup */
+	$('.menu .item')
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]: Estas 3 lineas probs should be 1

File: resources/views/user/details.blade.php [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+	$.ajaxSetup({
+		beforeSend: function(xhr) {
+			xhr.setRequestHeader('X-CSRF-TOKEN', $('meta[name="csrf-token"]').attr("content"));
+		}
+	});
+
+	/* courses */
+	let hasCourses = {!! $user->enrolledIn !!}.length > 0;
+
+	function showCourses() {
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]: Esta informaci&oacute;n la tienes desde que cargas la p&aacute;gina, hay alguna raz&oacute;n por la que esto no se hace via PHP?

File: resources/views/user/details.blade.php [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+	showCourses(); // execute on load
+
+	@if($user->id == Auth::user()->id)
+		/* course association workflow */
+		function validateCourseKey(courseKey) {
+			$("#courseKeyInputContainer").removeClass("error");
+			$("#courseKey").removeClass("error");
+
+			if(courseKey == "") {
+				$("#courseKeyInputContainer").addClass("error");
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]:
```suggestion
				$("#courseKeyInputContainer", "#courseKey").addClass("error");
```

Y borras la que sigue

File: resources/views/user/details.blade.php [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+			$('#no-courses-msg').show();
+		}
+	}
+
+	showCourses(); // execute on load
+
+	@if($user->id == Auth::user()->id)
+		/* course association workflow */
+		function validateCourseKey(courseKey) {
+			$("#courseKeyInputContainer").removeClass("error");
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]:
```suggestion
			$("#courseKeyInputContainer", "#courseKey").removeClass("error");
```

**Comment by hecerinc** [comment]
image

Probably should center the image or the name.

File: resources/views/user/details.blade.php [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+				@endif
+			</div>
+			<div class="twelve wide column">
+				<div class="ui top attached tabular menu">
+					<a class="active item" data-tab="projects">Projects</a>
+					<a class="item" data-tab="second">Second</a>
+					<a class="item" data-tab="courses">Courses</a>
+				</div>
+				<div class="ui bottom attached active tab segment" data-tab="projects">
+					@if(isset($user->projects) && count($user->projects) > 0)
```
Comment (hecerinc) [inline resources/views/user/details.blade.php]: Probably should open an issue to enable `user->projects`. No existe todav&iacute;a

File: resources/views/user/details.blade.php (hunk 5 above)
Comment (katiearriagam) [reply resources/views/user/details.blade.php]: Yep. es parte de los missing que vienen en la description del PR pero le pondré un TODO comment como quiera :) thanks!

**Comment by anakarenbm** [comment]
Cute UI ! 💌 

**Answering your question about project owners:**
No hay métodos para eso, pero @hecerinc is working on that en este [PR](https://github.com/ProyectoIntegrador2018/BlueMarket/pull/110/files)
va a ser algo así `$project->team->leader` o  `$project->team-> members `

**Pull request merged**